Sources/Testing/CMakeLists.txt

```diff
@@ -96,6 +96,14 @@ add_library(Testing
   Traits/Trait.swift)
 target_link_libraries(Testing PRIVATE
   _TestingInternals)
+if(NOT APPLE)
+  if(NOT CMAKE_SYSTEM_NAME STREQUAL WASI)
+    target_link_libraries(Testing PUBLIC
+      dispatch)
+  endif()
```

Comment:
@etcwilde what do you think of the generator expression way of writing this instead?

Reply:
eh, I think for this I'm fine keeping the explicit `if`.
Info on whether we should be linking dispatch into swift-testing seems like it should be coming from the Foundation configuration though, and then we wouldn't need it here at all. It would be part of the exposed interface from Foundation.